### DISCUSSION TITLE
React App in mirco ?

### DISCUSSION THREAD

**Discussion by ooralas** [deta/cloud-docs#245] [Help] [answered]
I just find out Deta, and I am really surprised how powerful Deta is.

**i would like to ask how to have a React app in a micros so i can show the React app on the home page. ?
has anyone done this bevor? or can I find documentation somewhere ?**

**Comment by abdelhai** [comment, marked as answer]
hi, thanks!

react apps are not well supported on micors yet. we're working on a better solution which could be ready in a few months.

**Comment by deathcrafter** [comment]
## If you are using node and express:
### Creating your React client:
- `create-react-app` your app in a directory starting with a dot e.g. `.client` inside of your deta micro directory. The command would be
```
npx create-react-app .client
```
*The reason to put the dot `.` infront of the directory name is that deta will ignore the directory when deploying.*
- Create an `.env` file inside the `.client` folder and add `BUILD_PATH='../client'` to it.
- Now build the client using
```
cd ./.client/ && npm run build && cd ..
```
- Then in your `index.js`, redirect to the react app:
```js
// install express with `npm install express`
const path = require("path");
const express = require("express");
const app = express();

app.use(express.static(path.resolve(__dirname, "client")));

app.get("*", (req, res) => res.sendFile(path.resolve(__dirname, "../client", "index.html")));

// export 'app'
module.exports = app;
```

### Deploying to Deta Micro:
Now you can use this command to always build the client and then deploy to the micro by using:
```
cd ./.client/ && npm run build && cd .. && deta deploy
```

**Reply by ooralas** [reply]
Thanks a lot, i will give it a try and write the result hier.

Actually i'm using Vite as a build-tool, but i think, that will work with also, i should refer to the `dist `directory.
